### PR TITLE
Added API for Redis set operations

### PR DESCRIPTION
**Pull request by rob-deans**
## How Has This Been Tested?

Made tests for both the String version of the function and the RedisString
## Checklist:

- [ ] I have submitted a [CLA form](https://github.com/IBM-Swift/CLA)
- [X] If applicable, I have updated the documentation accordingly.
- [X] If applicable, I have added tests to cover my changes.

### CODE REVIEW

File: Sources/Redis.swift [hunk 1]
```diff
@@ -1505,115 +1546,884 @@ public class Redis {
...
+        }
+    }
+    
+    ///
+    /// Determine if a given value is a member of a set
+    ///
+    /// - Parameter key: the String paramter for the key
+    /// - Parameter member: the String paramter for the member
+    ///
+    /// Returns: Integer reply: 1 if element is a member of the set,
```
Comment (shmuelk) [inline Sources/Redis.swift]: I think you should return a Bool not an Int. There is a redisBoolResponseHanlder that should take an Integer result and call a callback that takes a Bool.

File: Sources/Redis.swift [hunk 2]
```diff
@@ -1505,115 +1546,884 @@ public class Redis {
...
+        }
+    }
+    
+    ///
+    /// Determine if a given value is a member of a set
+    ///
+    /// - Parameter key: the RedisString paramter for the key
+    /// - Parameter member: the RedisString paramter for the member
+    ///
+    /// Returns: Integer reply: 1 if element is a member of the set,
```
Comment (shmuelk) [inline Sources/Redis.swift]: I think you should return a Bool not an Int. There is a redisBoolResponseHanlder that should take an Integer result and call a callback that takes a Bool.

File: Sources/Redis.swift [hunk 3]
```diff
@@ -1505,115 +1546,884 @@ public class Redis {
...
+    }
+    
+    ///
+    /// Move a member from one set to another
+    ///
+    /// - Parameter source: the Source set from where to move the member from
+    /// - Parameter destination: the Destination set from where to move the member to
+    /// - Parameter member: the String parameter for the member to be moved
+    ///
+    /// Returns: Integer reply: 1 if element is moved,
```
Comment (shmuelk) [inline Sources/Redis.swift]: I think you should return a Bool not an Int. There is a redisBoolResponseHanlder that should take an Integer result and call a callback that takes a Bool.

File: Sources/Redis.swift [hunk 4]
```diff
@@ -1505,115 +1546,884 @@ public class Redis {
...
+    }
+    
+    ///
+    /// Move a member from one set to another
+    ///
+    /// - Parameter source: the Source set from where to move the member from
+    /// - Parameter destination: the Destination set from where to move the member to
+    /// - Parameter member: the RedisString parameter for the member to be moved
+    ///
+    /// Returns: Integer reply: 1 if element is moved,
```
Comment (shmuelk) [inline Sources/Redis.swift]: I think you should return a Bool not an Int. There is a redisBoolResponseHanlder that should take an Integer result and call a callback that takes a Bool.

File: Tests/SwiftRedisTests/TestSetCommandsPart2.swift [hunk 5]
```diff
@@ -0,0 +1,1150 @@
...
+            
+            redis.sadd(self.key1, members: self.member1, self.member4) {
+                (retrievedTotalElementAdded: Int?, error: NSError?) in
+                
+                XCTAssertNil(error)
+                XCTAssertNotNil(retrievedTotalElementAdded)
+                XCTAssertEqual(retrievedTotalElementAdded, 1)
+                
+            }
+            expectation1.fulfill()
```
Comment (shmuelk) [inline Tests/SwiftRedisTests/TestSetCommandsPart2.swift]: The tests in general seekm ok, with major issue, that perhaps we can solve later on.

The tests assume that the Kitura-redis APIs run synchronously. While this is true now, we will eventually change that. When that happens all of these tests will break as the line of code expectation1.fulfill() might run before any of the callbacks would run.

I think the series of tests must be run as a nested sequence. The second one invoked from the callback of the first one, the third one invoked from the callback of the second one, etc. In the end of the callback of the last test in the test group, call fulfill on the expectation.

Comment (rob-deans) [reply Tests/SwiftRedisTests/TestSetCommandsPart2.swift]: @shmuelk Should I make those expectation changes or leave it for now?

**Comment by rob-deans** [comment]
@shmuelk Can you please review this again?

**Pull request merged**